x-pack/plugins/maps/public/actions/map_actions.ts

```diff
@@ -295,7 +284,13 @@ export function setQuery({
       searchSessionMapBuffer: getSearchSessionMapBuffer(getState()),
     };
 
-    if (_.isEqual(nextQueryContext, prevQueryContext)) {
+    if (
+      !forceRefresh &&
+      _.isEqual(nextQueryContext, {
```

Comment:
How about removing `forceRefreshTriggeredFromGlobalQueryTime` from `nextQueryContext`? Its making this if statement overly complicated. Then you can just add `forceRefreshTriggeredFromGlobalQueryTime` into dispatch of SET_QUERY.